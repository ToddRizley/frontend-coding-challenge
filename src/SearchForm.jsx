import React, { Component } from 'react';
export default class SearchForm extends Component {

  render(){
    return(
      <div className="searchbar">
        <input
          onChange={this.props.searchByTitle.bind(this)}
          ref='searchBar'
          type='text'
          placeholder='Search by event title'
        />
        <div className='fa fa-search' />
      </div>
  )};
};
