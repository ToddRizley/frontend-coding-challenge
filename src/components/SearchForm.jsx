import React, { Component } from 'react';

export default class SearchForm extends Component {

  render(){
    return(
    <form className='search-form col-xs-6'>
      <div className="form-group has-feedback">
        <label className="control-label">Search Events By Title</label>
        <input
          onChange={this.props.searchByTitle.bind(this)}
          type='text'
          className="form-control"
          placeholder='Search by event title'
        />
      <i className="glyphicon glyphicon-search form-control-feedback"></i>
      </div>
    </form>
  )};
};
