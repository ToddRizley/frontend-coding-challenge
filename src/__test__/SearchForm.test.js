import React from 'react';
import SearchForm from '../components/SearchForm'
import renderer from 'react-test-renderer';

describe('<SearchForm/>', () => {
  var searchByTitle = jest.fn()
   it('renders correctly', () => {
    const rendered = renderer.create(
      <SearchForm searchByTitle={searchByTitle}/>
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
