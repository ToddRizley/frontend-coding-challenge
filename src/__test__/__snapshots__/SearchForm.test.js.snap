// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchForm/> renders correctly 1`] = `
<form
  className="search-form col-xs-6"
>
  <div
    className="form-group has-feedback"
  >
    <label
      className="control-label"
    >
      Search Events By Title
    </label>
    <input
      className="form-control"
      onChange={[Function]}
      placeholder="Search by event title"
      type="text"
    />
    <i
      className="glyphicon glyphicon-search form-control-feedback"
    />
  </div>
</form>
`;
